src/mapper.rs

```diff
@@ -0,0 +1,170 @@
+use std::collections::HashMap;
+use std::{borrow::Cow, fmt::Write};
+
+use crate::mapping::{parse_mapping_line, MappingRecord};
+use crate::stacktrace::{parse_stacktrace_line, StackFrame};
+
+#[derive(Debug)]
+struct MemberMapping<'s> {
+    startline: usize,
+    endline: usize,
+    original_class: Option<&'s str>,
+    original: &'s str,
+    original_startline: usize,
+    original_endline: Option<usize>,
+}
+
+#[derive(Debug)]
+struct ClassMapping<'s> {
+    original: &'s str,
+    obfuscated: &'s str,
+    members: HashMap<&'s str, Vec<MemberMapping<'s>>>,
+}
+
+/// A Proguard Remapper.
+///
+/// This can remap frames one at a time, or the complete raw stacktrace.
+#[derive(Debug)]
+pub struct Mapper<'s> {
+    classes: HashMap<&'s str, ClassMapping<'s>>,
+}
+
+impl<'s> Mapper<'s> {
+    /// Create a new Proguard Remapper.
+    pub fn new(mapping: &'s str) -> Self {
```

Comment:
How would you feel about still exposing the `MappingRecord` and having a `ProguardMapping<'s>` type that allows iteration over the records. That way, someone who wants to inspect the files can still do it.